rotkehlchen/chain/evm/decoding/thegraph/decoder.py

```diff
@@ -0,0 +1,185 @@
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        log.info(delegator)
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        tokens_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+        lock_timeout_secs = hex_or_bytes_to_int(context.tx_log.data[64:128])
+        tokens_amount_norm = token_normalized_value(
+            token_amount=tokens_amount,
+            token=self.token,
+        )
+        log.info(tokens_amount_norm)
+        # create a new informational event about undelegation and the expiring lock on tokens
+        event = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=self.token,
+            balance=Balance(),
+            location_label=delegator,
+            notes=(
+                f'Undelegate {tokens_amount_norm} GRT from indexer {indexer}.'
+                f' Lock expires in {lock_timeout_secs} seconds'
+            ),
+            counterparty=CPT_THEGRAPH,
+            address=context.tx_log.address,
+        )
+        log.info(event)
+        return DecodingOutput(event=event)
+
+    def _decode_stake_withdrawn(self, context: DecoderContext) -> DecodingOutput:
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        tokens_amount_norm = token_normalized_value(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token=self.token,
+        )
+        # create action item that will modify the relevant Transfer event that will appear later
+        action_item = ActionItem(
+            action='transform',
+            from_event_type=HistoryEventType.RECEIVE,
+            from_event_subtype=HistoryEventSubType.NONE,
+            asset=self.token,
+            amount=tokens_amount_norm,
+            to_event_type=HistoryEventType.STAKING,
+            to_event_subtype=HistoryEventSubType.REMOVE_ASSET,
+            to_notes=f'Withdraw {tokens_amount_norm} GRT from indexer {indexer}',
+            to_counterparty=CPT_THEGRAPH,
+        )
+        return DecodingOutput(action_items=[action_item])
+
+    # -- DecoderInterface methods
+
+    @staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (THEGRAPH_CPT_DETAILS,)
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            self.staking_contract: (self._decode_delegator_staking,),
+        }
```

Comment:
if it fits in pne line keep it in one line

```suggestion
        return {self.staking_contract: (self._decode_delegator_staking,)}
```